mbed/mbed.py

```diff
@@ -2221,6 +2231,23 @@ def compile_(toolchain=None, target=None, profile=False, compile_library=False,
                   + (['-v'] if verbose else [])
                   + args,
                   env=env)
+            
+            if flash:
+                fw_name = artifact_name if artifact_name else program.name
+                fw_fbase = os.path.join(build_path, fw_name)
+                fw_file = fw_fbase + ('.hex' if os.path.exists(fw_fbase+'.hex') else '.bin')
```

Comment:
You only check that the `.hex` file exists. I think you should check for `.bin` as well.